frontend/src/metabase/embedding-sdk/components/SimpleDataPicker/SimpleDataPicker.tsx

```diff
@@ -1,108 +1,94 @@
-import { useState } from "react";
-import { t } from "ttag";
+import { useDisclosure } from "@mantine/hooks";
+import { type ReactNode, useMemo } from "react";
 
-import EmptyState from "metabase/components/EmptyState";
-import { CONTAINER_WIDTH } from "metabase/query_builder/components/DataSelector/constants";
-import { Flex, Icon, NavLink, Paper, ScrollArea, TextInput } from "metabase/ui";
-import type { TableId } from "metabase-types/api";
+import { useSearchQuery } from "metabase/api";
+import { isModel } from "metabase/browse/models/utils";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import { Box, Popover } from "metabase/ui";
+import { getQuestionVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
+import type { SearchResult, TableId } from "metabase-types/api";
+import { SortDirection } from "metabase-types/api/sorting";
+
+import { SimpleDataPickerView } from "./SimpleDataPickerView";
 
 interface SimpleDataPickerProps {
+  filterByDatabaseId: number | null;
   selectedEntity?: TableId;
-  options: Options[];
-  onClick: (option: any) => void;
-}
-
-interface Options {
-  id: number;
-  display_name: string;
+  isInitiallyOpen: boolean;
+  triggerElement: ReactNode;
+  setSourceTableFn: (tableId: TableId) => void;
 }
 
-const TEN_OPTIONS_HEIGHT = 10 * 33;
-
 export function SimpleDataPicker({
+  filterByDatabaseId,
   selectedEntity,
-  options,
-  onClick,
+  isInitiallyOpen,
+  setSourceTableFn,
+  triggerElement,
 }: SimpleDataPickerProps) {
-  const shouldShowSearchBar = options.length > 10;
-  const [searchText, setSearchText] = useState("");
-  function filterSearch(option: Options): boolean {
-    if (searchText) {
-      return normalizeString(option.display_name).includes(
-        normalizeString(searchText),
-      );
-    }
+  const [isDataPickerOpened, { toggle, close }] =
+    useDisclosure(isInitiallyOpen);
+  const { data, isLoading, error } = useSearchQuery({
```

Comment:
nit: I prefer aliasing `data` to make it a little easier to read 

```suggestion
  const { data: searchQuery, isLoading, error } = useSearchQuery({
```